activestorage/lib/active_storage/attached/changes/create_one.rb

```diff
@@ -22,8 +22,12 @@ def blob
 
     def upload
       case attachable
-      when ActionDispatch::Http::UploadedFile, Rack::Test::UploadedFile
+      when ActionDispatch::Http::UploadedFile
         blob.upload_without_unfurling(attachable.open)
+      when Rack::Test::UploadedFile
+        blob.upload_without_unfurling(
+          attachable.respond_to?(:open) ? attachable.open : attachable # or attachable.reopen
```

Comment:
I think you need to pick if you want to use `#open` or `#reopen`, and then remove the comment.